atlite/wind.py

```diff
@@ -131,49 +135,48 @@ def download_turbineconf(turbine=None, store_locally=True):
         Has the same format as returned by 'atlite.ressource.get_turbineconf(name)'.
     """
     
-    try: 
-        # Get the turbine list
-        OEDB_URL = 'https://openenergy-platform.org/api/v0/schema/supply/tables/turbine_library/rows'
-        result = requests.get(OEDB_URL)
-    except requests.exceptions.RequestException as e:
-        logger.info(f"Connection to OEDB failed with:\n\n{str(e)}")
-        return None
+    OEDB_URL = 'https://openenergy-platform.org/api/v0/schema/supply/tables/turbine_library/rows'
 
-    # Convert JSON to dataframe for easier filtering
-    # Only consider turbines with power curves available
-    df = pd.DataFrame.from_dict(result.json())
-    df = df[df.has_power_curve]
+    # Cache turbine request locally
+    global _oedb_turbines
 
-    # Check which information is provided for lookup and proceed accordingly
-    if turbine.get('id'):
-        logger.info(f"Searching for turbine in OEDB with id '{turbine['id']}'.")
-        ds = df[df.id == int(turbine['id'])]
-        
-    else:
-        if not all({isinstance(v, str)
-                    for k,v in turbine.items() if k in ['name', 'manufacturer']}
-                  ):
-            
-            logger.error("'name' and 'manufacturer' must be provided. "
-                         "Alternatively provide a turbine id from OEDB.")
+    if _oedb_turbines is None:
+        try: 
+            # Get the turbine list
+            result = requests.get(OEDB_URL)
+        except requests.exceptions.RequestException as e:
+            logger.info(f"Connection to OEDB failed with:\n\n{str(e)}")
             return None
-        logger.info(f"Searching turbine power curve in OEDB database using "
-                    f"manufacturer '{turbine['manufacturer']}'"
-                    f"and name '{turbine['name']}'.")
-            
-        ds = df[df.name == turbine['name']]
-        ds = ds[ds.manufacturer == turbine['manufacturer']]
 
-    if len(ds) < 1 :
-        logger.info(f"No turbine found for {turbine}.")
+        # Convert JSON to dataframe for easier filtering
+        # Only consider turbines with power curves available
+        df = pd.DataFrame.from_dict(result.json())
+        _oedb_turbines = df[df.has_power_curve]
+
+
+    logger.info("Searching turbine power curve in OEDB database using " +
+                ", ".join(f"{k}='{v}'" for (k,v) in turbine.items()) + ".")
+    # Working copy
+    df = _oedb_turbines
+    if turbine.get('id'):
+        df = df[df.id == int(turbine['id'])]
+    if turbine.get('name'):
+        df = df[df.name == turbine['name']]
+    if turbine.get('manufacturer'):
+        df = df[df.manufacturer == turbine['manufacturer']]
```

Comment:
I'd also support other fields like `wind_zone_dibt` and `wind_class_iec`, probably with a `contains` check instead:
```python
for field in ('wind_zone_dibt', 'wind_class_iec'):
    if turbine.get(field):
        df = df[df[field].str.contains(turbine[field])]
```

Reply:
Ok, well. The examples do make it clearer, thanks!
I think it is still too much feature and too less simple, but if you insist... .

It should be possible now to use the example arguments you provided above.

I would not implement the selection based on other fields at the moment: The OEDB is not well enough curated at that point and it would be easier to improve the quality of their DB than writing a working parser.

Reply:
Ok, future problem, then!